lib/Dialect/FIRRTL/Transforms/GrandCentral.cpp

```diff
@@ -411,10 +411,60 @@ struct VerbatimBuilder {
   struct Base {
     SmallString<128> string;
     SmallVector<Attribute> symbols;
+    SmallVector<Value, 2> subst;
     VerbatimBuilder builder() { return VerbatimBuilder(*this); }
     operator VerbatimBuilder() { return builder(); }
   };
 
+  /// Increment all the indices inside `{{`, `}}` by one. This is to indicate
+  /// that a value is added to the `substitutions` of the verbatim op, other
+  /// than the symbols.
+  void incrementIds() {
+    StringRef begin = "{{";
+    StringRef end = "}}";
+    // The replacement string.
+    size_t from = 0;
+    replStr.clear();
+    while (from < base.string.size()) {
+      // Search for the first `{{` and `}}`.
+      auto beginAt = base.string.find(begin, from);
+      auto endAt = base.string.find(end, beginAt);
+      // Copy the string as is, until the `{{`.
+      replStr.append(base.string.substr(from, beginAt - from));
```

Comment:
I think this works when `beginAt` is `npos`, but only because of the underlying type (size_t) and value of `npos` (`~size_t(0)`) and the very-likely-to-be-true assumption that `npos - from` is larger than remaining characters remaining.

Anyway, maybe touch this up to avoid this?